### ISSUE TITLE
UnusedPrivateMembers on expect class

### ISSUE THREAD

**Issue by 4face-studi0** [detekt/detekt#2636]
## Expected Behavior
Function members **should not** be reported for `except class`

## Observed Behavior
Functions members are reported for `expect class`

## Steps to Reproduce
// _DetektTest.kt
```kotlin
expect class DetektTest  {
    fun test(i: Int)
}
```
* write `actual` declarations that **uses** the param `i`
* run Detekt on CLI and notice the following report:
`UnusedPrivateMember - [i] at ~/_DetektTest.kt:2:14`

## Context
none

## Your Environment
* Version of detekt used: **1.8.0**
* Version of Gradle used (if applicable): **6.3 stable**
* Operating System and version: **macOS 10.15.4 (19E287)**
* Link to your project (if it's a public repository): https://github.com/4face-studi0/Covid19

**Comment by mikesamuel** [comment]
I also see this behavior for formal parameters to an expected class constructor or expected function.

```kotlin
expect class Foo(x: Int, y:Int)

expect fun foo(x: Int, y: Int)
```

Also on release 1.8.0 I get

     style - 20min debt
            UnusedPrivateMember - [x] at .../Foo.kt:14:16
            UnusedPrivateMember - [y] at .../Foo.kt:14:24
            UnusedPrivateMember - [x] at .../Foo.kt:12:18
            UnusedPrivateMember - [y] at .../Foo.kt:12:26

I can workaround via `@Suppress("UnusedPrivateMember")` on the `expect` declarations.
